lldb/tools/lldb-dap/DAP.cpp

```diff
@@ -1289,13 +1293,108 @@ protocol::Capabilities DAP::GetCustomCapabilities() {
 }
 
 void DAP::StartEventThread() {
-  event_thread = std::thread(&DAP::EventThread, this);
+  // Get event thread for this debugger (creates it if it doesn't exist).
+  event_thread_sp = DAPSessionManager::GetInstance().GetEventThreadForDebugger(
+      debugger, this);
 }
 
 void DAP::StartProgressEventThread() {
   progress_event_thread = std::thread(&DAP::ProgressEventThread, this);
 }
 
+void DAP::StartEventThreads() {
+  if (clientFeatures.contains(eClientFeatureProgressReporting))
+    StartProgressEventThread();
+
+  StartEventThread();
+}
+
+llvm::Error DAP::InitializeDebugger(std::optional<int> debugger_id,
+                                    std::optional<lldb::user_id_t> target_id) {
+  // Validate that both debugger_id and target_id are provided together.
+  if (debugger_id.has_value() != target_id.has_value()) {
+    return llvm::createStringError(
+        "Both debuggerId and targetId must be specified together for debugger "
+        "reuse, or both must be omitted to create a new debugger");
+  }
+
+  // Initialize debugger instance (shared or individual).
+  if (debugger_id && target_id) {
+    // Find the existing debugger by ID
+    debugger = lldb::SBDebugger::FindDebuggerWithID(*debugger_id);
+    if (!debugger.IsValid()) {
+      return llvm::createStringError(
+          "Unable to find existing debugger for debugger ID");
+    }
```

Comment:
this doesn't look very clean.
What about creating two `InitializeDebugger` functions? One doesn't take parameters and the other one takes forcefully those two ids?